TypeScriptDefinitions/async.d.ts

```diff
@@ -0,0 +1,165 @@
+// Type definitions for Async 1.4.2
```

Comment:
Yes, for now this directory should be submitted.  These are TypeScript definition files that aid Webstorm in intellisense and error checking and get updated when you update dependencies in package.json.  They are discussed in my upcoming node guide.  I'm hoping to improve things in the long term so that these will be handled slightly differently (and possibly not submitted) but for now this is the same thing we do in all of our other modules.